### PR TITLE
Update Rails to 6.1

### PR DESCRIPTION
**Pull request by thalesmiguel**
**What this PR does / why we need it**:

This PR attempts to update Porta's Rails version to 6.1.x. It also tries to update the minimum amount of gems necessary for the Rails update.

**Which issue(s) this PR fixes** 

[THREESCALE-7405](https://issues.redhat.com/browse/THREESCALE-7405)

**Verification steps** 

That's a good question. Put it on `staging` and/or the development cluster and try to break it? 

**Special notes for your reviewer**:

When I first attempted to update Rails to 6.0.x, I noticed that there shouldn't be too much of a difference compared with the update to 6.1.x, therefore I pushed for a bigger update.
Also, 6.0.x's EOL will happen in June 1st, 2023.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -136,8 +136,10 @@ gem 'rack-utf8_sanitizer'
 gem 'jwt', '~> 1.5.2', require: false
 
 group :assets do
-  gem 'codemirror-rails', '~> 5.6'
-  gem 'font-awesome-rails', '~> 4.7.0.5'
+  # codemirror-rails is deprecated and won't be updated to be used on Rails 6+
+  # check https://github.com/fixlr/codemirror-rails/issues/61
+  gem 'codemirror-rails', git: 'https://github.com/thalesmiguel/codemirror-rails.git', branch: 'master'
```
Comment (thalesmiguel) [inline Gemfile:141]: [This is definitely not cool and needs to be changed]

`codemirror-rails` won't receive more updates, including the ability to run on Rails 6.0 or superior.
To try and run this, I forked the repo (to my personal account) and enabled its use for Rails 6.1.

A couple options to make it better:
- Take this chance to open another PR that removes `codemirror-rails` and, instead, imports whatever's needed for codemirror to work from a JS package;
  - This also means that the ruby code that calls this gem will need to be removed;
- Fork the repo into the `3scale` namespace and perform the same changes I did so we don't need to worry about removing the gem for now.
  - Seems like this was already done in the past with `activejob-uniqueness`.

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += %i[activation_code cms_token credit_card credit_card_auth_code
-                                                 credit_card_authorize_net_payment_profile_token credit_card_expires_on
-                                                 credit_card_partial_number crypted_password janrain_api_key lost_password_token
-                                                 password password_digest payment_gateway_options payment_service_reference salt
-                                                 site_access_code sso_key user_key access_token service_token provider_key app_key]
+Rails.application.config.filter_parameters += %i[
+  _key access_token activation_code app_key certificate cms_token credit_card credit_card_auth_code
```
Comment (mayorova) [inline config/initializers/filter_parameter_logging.rb:5]: `_key` ?

Comment (thalesmiguel) [reply config/initializers/filter_parameter_logging.rb:5]: Yes. It came with `rails app:update`.

Comment (mayorova) [reply config/initializers/filter_parameter_logging.rb:5]: OK, I thought it was just reordering, and `_key` was a piece of `user_key` or something :upside_down_face:

File: config/storage.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+test:
```
Comment (mayorova) [inline config/storage.yml:1]: Is this file required?
I just removed it here: https://github.com/3scale/porta/pull/3225/files#diff-188abe2a4493506d81577fbbee0b279b889e59f50d509402fbbfb620c8e58780

I think we can bring it back when we migrate from Paperclip to ActiveStorage

Comment (mayorova) [reply config/storage.yml:1]: Ignore me, I can see it is required in Rails 6... OK :upside_down_face:

**Comment by akostadinov** [comment]
I would suggest going one by one anyway unless you see any harm in that.

I've copied some checkpoints from the 5.2 upgrade to https://issues.redhat.com/browse/THREESCALE-7405 to keep our approach consistent.

**Comment by thalesmiguel** [comment]
> I would suggest going one by one anyway unless you see any harm in that.

@akostadinov By "one by one", do you mean the Rails version update? Like in "Let's first go to 6.0.x and then we proceed to update to 6.1.x?".
If so, I checked the changes and update guides from 6.0 to 6.1 and I think it'll be safe to go all the away.

My idea is to proceed with 6.1.x as the end goal and, if I see something that'll prevent the updated, I'll go back to 6.0.x and add comments for the future updated.
Also, as mentioned in the PR's description, 6.0.x's EOL will happen in June 1st, 2023 already.

File: Gemfile (hunk 1 above)
Comment (akostadinov) [reply Gemfile:141]: If is the other way around how we did previous updates. We try to update incompatible components while on the old Rails version so we don't have a huge PR at the end.

So if it is feasible to get rid of unsupported gem, that would be preferable in a pull request coming before the rails update one.

**Comment by thalesmiguel** [comment]
Closed in favour of https://github.com/3scale/porta/pull/3311
